server/sonar-server/src/main/java/org/sonar/server/qualityprofile/BuiltInQualityProfilesNotification.java

```diff
@@ -0,0 +1,142 @@
+
+  private final List<Profile> profiles = new ArrayList<>();
+
+  public BuiltInQualityProfilesNotification addProfile(Profile profile) {
+    profiles.add(profile);
+    return this;
+  }
+
+  public Notification serialize() {
+    Notification notification = new Notification(BUILT_IN_QUALITY_PROFILES);
+    notification.setFieldValue(NUMBER_OF_PROFILES, String.valueOf(profiles.size()));
+    AtomicInteger count = new AtomicInteger();
+    profiles.forEach(profile -> {
+      int index = count.getAndIncrement();
+      notification.setFieldValue(index + PROFILE_NAME, profile.getProfileName());
+      notification.setFieldValue(index + LANGUAGE, profile.getLanguage());
+      notification.setFieldValue(index + NEW_RULES, String.valueOf(profile.getNewRules()));
+    });
+    return notification;
+  }
+
+  public static BuiltInQualityProfilesNotification parse(Notification notification) {
+    checkState(BUILT_IN_QUALITY_PROFILES.equals(notification.getType()),
+      "Expected notification of type %s but got %s", BUILT_IN_QUALITY_PROFILES, notification.getType());
+    BuiltInQualityProfilesNotification notif = new BuiltInQualityProfilesNotification();
+    String numberOfProfilesText = notification.getFieldValue(NUMBER_OF_PROFILES);
+    checkState(numberOfProfilesText != null, "Could not read the built-in quality profile notification");
+    Integer numberOfProfiles = Integer.valueOf(numberOfProfilesText);
+    IntStream.rangeClosed(0, numberOfProfiles - 1)
+      .mapToObj(index -> Profile.newBuilder(
+        getNonNullFieldValue(notification, index + PROFILE_NAME),
+        getNonNullFieldValue(notification, index + LANGUAGE))
+        .setNewRules(parseInt(getNonNullFieldValue(notification, index + NEW_RULES)))
+        .build())
+      .forEach(notif::addProfile);
+    return notif;
+  }
+
+  private static String getNonNullFieldValue(Notification notification, String key) {
+    String value = notification.getFieldValue(key);
+    return requireNonNull(value, format("Notification field '%s' is null", key));
+  }
+
+  public List<Profile> getProfiles() {
+    return profiles;
+  }
+
+  public static class Profile {
+    private final String profileName;
+    private final String language;
+    private final int newRules;
+
+    public Profile(Builder builder) {
+      this.profileName = builder.profileName;
+      this.language = builder.language;
+      this.newRules = builder.newRules;
+    }
+
+    public String getProfileName() {
+      return profileName;
+    }
+
+    public String getLanguage() {
+      return language;
+    }
+
+    public int getNewRules() {
+      return newRules;
+    }
+
+    public static Builder newBuilder(String profileName, String language) {
+      return new Builder(profileName, language);
+    }
+
+    public static class Builder {
+      private final String profileName;
+      private final String language;
+      private int newRules;
+      private int removedRules;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "removedRules" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1068)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the "removedRules" field and declare it as a local variable in the relevant methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1450)